src/claim_canister/tests/claim_tests.rs

```diff
@@ -0,0 +1,402 @@
+mod setup;
+use candid::{decode_one, encode_args, Nat, Principal};
+use claim_canister::types::{ClaimError, ClaimInfo, ClaimStatus};
+use claim_canister::TIMELOCK_DURATION;
+use setup::setup;
+use std::time::Duration;
+
+#[test]
+fn test_claim_positive_flow() {
```

Comment:
Also, add to the test that the receiver's principal actually receives the expected amount